feat: add `slice` method to `array/complex64` (#1257)
stdlib-js/stdlib: lib/node_modules/@stdlib/array/complex64/README.md

-   If a target array cannot accommodate all values (i.e., the length of source array plus `i` exceeds the target array length), the method throws an error.
-   If provided a [typed array][@stdlib/array/typed] which shares an [`ArrayBuffer`][@stdlib/array/buffer] with the target array, the method will intelligently copy the source range to the destination range.

<a name="method-slice"></a>

> #### Complex64Array.prototype.slice( \[begin\[, end]] )

Reviewer: ```suggestion
#### Complex64Array.prototype.slice( \[start\[, end]] )
```

Use consistent naming conventions (see TypeScript and main.js).

> Returns a new typed array containing the extracted elements.

Reviewer: ```suggestion
Copies a portion of a typed array to a new typed array.
```

```javascript
var realf = require( '@stdlib/complex/realf' );
var imagf = require( '@stdlib/complex/imagf' );

var arr = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );

> var slicearr = arr.slice();
> // returns <Complex64Array>
> 
> var len = slicearr.length;
> // returns 4
> 
> var z = slicearr.get( 0 );
> // returns <Complex64>
> 
> var re = realf( z );
> // returns 1.0
> 
> var im = imagf( z );
> // returns 2.0
> 
> z = slicearr.get( len-1 );

Reviewer: ```suggestion
var out = arr.slice();
// returns <Complex64Array>

var len = out.length;
// returns 4

var z = out.get( 0 );
// returns <Complex64>

var re = realf( z );
// returns 1.0

var im = imagf( z );
// returns 2.0

z = out.get( len-1 );
```

By convention, we typically prefer simple and consistent variable names (e.g., `x`, `y`, `z`, `out`, `fcn`, `f`, `arr`, etc).

// returns <Complex64>

re = realf( z );
// returns 7.0

im = imagf( z );
// returns 8.0
```

> By default, the method returns a typed array beginning with the first array element. To specify an alternative array index at which to begin, provide a `begin` index (inclusive).

Reviewer: ```suggestion
By default, the method returns a typed array beginning with the first array element. To specify an alternative array index at which to begin, provide a `start` index (inclusive).
```

```javascript
var imagf = require( '@stdlib/complex/imagf' );
var realf = require( '@stdlib/complex/realf' );

var arr = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );

> var slicearr = arr.slice( 1 );
> // returns <Complex64Array>
> 
> var len = slicearr.length;
> // returns 3
> 
> var z = slicearr.get( 0 );

Reviewer: ```suggestion
var out = arr.slice( 1 );
// returns <Complex64Array>

var len = out.length;
// returns 3

var z = out.get( 0 );
```

// returns <Complex64>

var re = realf( z );
// returns 3.0

var im = imagf( z );
// returns 4.0
```

> By default, the method returns a typed array which includes all array elements after `begin`. To limit the number of array elements after `begin`, provide an `end` index (exclusive).

Reviewer: ```suggestion
By default, the method returns a typed array which includes all array elements after `start`. To limit the number of array elements after `start`, provide an `end` index (exclusive).
```

```javascript
var realf = require( '@stdlib/complex/realf' );
var imagf = require( '@stdlib/complex/imagf' );

var arr = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );

> var slicearr = arr.slice( 1, -1 );
> // returns <Complex64Array>
> 
> var len = slicearr.length;
> // returns 2
> 
> var z = slicearr.get( 0 );
> // returns <Complex64>
> 
> var re = realf( z );
> // returns 3.0
> 
> var im = imagf( z );
> // returns 4.0
> 
> z = slicearr.get( len-1 );

Reviewer: ```suggestion
var out = arr.slice( 1, -1 );
// returns <Complex64Array>

var len = out.length;
// returns 2

var z = out.get( 0 );
// returns <Complex64>

var re = realf( z );
// returns 3.0

var im = imagf( z );
// returns 4.0

z = out.get( len-1 );
```
